server/routes.js

```diff
@@ -0,0 +1,166 @@
+var db = require('./db')
+
+module.exports = {
+  getDonors: getDonors,
+  getRecipients: getRecipients,
+  getDonor: getDonor,
+  getRecipient: getRecipient,
+  updateTicket: updateTicket,
+  updateComment: updateComment,
+  getTickets: getTickets,
+  addTicket: addTicket,
+  getDonorTicket: getDonorTicket,
+  getRecipientTicket: getRecipientTicket,
+  getTicketComments: getTicketComments
+}
+
+function getDonors(req, res) {
+  db.getDonors()
+    .then(function (donors) {
+      res.json(donors)
+    })
+    .catch(function (err) {
+      res.send(err.message).status(500)
+    })
+}
+
+function getRecipients(req, res) {
+  db.getRecipients()
+    .then(function (recipients) {
+      res.json(recipients)
+    })
+    .catch(function (err) {
+      res.send(err.message).status(500)
+    })
+}
+
+function getDonor(req, res) {
+  var donorId = req.params.id
+  db.getDonor(donorId)
+    .then(function (donor) {
+      res.json(donor)
+    })
+    .catch(function (err) {
```

Comment:
There's no checks to ensure the donor actually exists here! You need to look at the size of the result set you get back, and if there's no donor issue a 404 because the resource doesn't exist.